middleware.ts

```diff
@@ -0,0 +1,29 @@
+import { createMiddlewareClient } from "@supabase/auth-helpers-nextjs"
```

Comment:
> About Next.js, and I'm still getting used to the whole server/client approach in the same place. Sometimes I wonder if a file will run server or client side, or at build time or run time, etc. The errors most of the time are helpful though, and the community has most of the answers.

Based on its docs, every file under `/app` is considered a SSC unless you use `use client`. [resource](https://nextjs.org/docs/app/building-your-application/routing#the-app-router). I also liked this [When to use server and client components](https://nextjs.org/docs/getting-started/react-essentials#when-to-use-server-and-client-components) diagram (and the whole page in general)

> Trying to set cookies/handle auth manually. Looking back I think I was just not using the library I chose for the task (supabase).
> SSR and hydration. Using styled components apparently requires a pretty good understanding of these two, for now I think I'll opt for not styling that way.